src/stores/SpaceStore.tsx

```diff
@@ -124,6 +125,41 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
         return this._suggestedRooms;
     }
 
+    public async setActiveRoomInSpace(space: Room | null) {
+        if (space && !space.isSpaceRoom()) return;
+        if (space !== this.activeSpace) await this.setActiveSpace(space);
+
+        if (space) {
+            const notificationState = this.getNotificationState(space.roomId);
+            const roomId = notificationState.getFirstRoomWithNotifications();
+            defaultDispatcher.dispatch({
+                action: "view_room",
+                room_id: roomId,
+                context_switch: true,
+            });
+        } else {
+            const lists = RoomListStore.instance.unfilteredLists;
+            TAG_ORDER.every(t => {
```

Comment:
hmm, what is this trying to do? The code looks like it's running a loop but it returns a boolean (as required by the `every` contract), but that returned boolean is never saved. Is this meant to be a `forEach` instead?

As a point of pure preference, a real `for` loop is better over `forEach` largely to ensure clarity.

Reply:
No, the reason that I've used `.every()`is because it exits the loop as soon the loop returns false(when an unread room is found) which cannot be done by forEach. I'll change it to use for loop.